ambry-cloud/src/main/java/com.github.ambry.cloud/azure/CosmosChangeFeedFindToken.java

```diff
@@ -0,0 +1,239 @@
+  }
+
+  /**
+   * Constructor to create a {@link CosmosChangeFeedFindToken} with specified token values and specified version.
+   * @param bytesRead bytes read by remote so far.
+   * @param startContinuationToken start token from Cosmos.
+   * @param endContinuationToken end token from Cosmos.
+   * @param index index in cache upto which items are consumed by remote.
+   * @param totalItems total number of items in cache.
+   * @param cacheSessionId request id of the change feed query.
+   * @param version token version.
+   */
+  public CosmosChangeFeedFindToken(long bytesRead, String startContinuationToken, String endContinuationToken,
+      int index, int totalItems, String cacheSessionId, short version) {
+    this.version = version;
+    this.bytesRead = bytesRead;
+    this.startContinuationToken = startContinuationToken;
+    this.endContinuationToken = endContinuationToken;
+    this.index = index;
+    this.totalItems = totalItems;
+    this.cacheSessionId = cacheSessionId;
+  }
+
+  /**
+   * Deserialize {@link CosmosChangeFeedFindToken} object from input stream.
+   * @param inputStream {@link DataOutputStream} to deserialize from.
+   * @return {@link CosmosChangeFeedFindToken} object.
+   * @throws IOException
+   */
+  public static CosmosChangeFeedFindToken fromBytes(DataInputStream inputStream) throws IOException {
+    CosmosChangeFeedFindToken cosmosChangeFeedFindToken;
+    DataInputStream stream = new DataInputStream(inputStream);
+    short version = stream.readShort();
+    switch (version) {
+      case VERSION_0:
+        FindTokenType type = FindTokenType.values()[stream.readShort()];
+        if (type != FindTokenType.CloudBased) {
+          throw new IllegalArgumentException(
+              String.format("Invalid token type %s found while deserialization. Expected %s.", type,
+                  FindTokenType.CloudBased));
+        }
+        long bytesRead = stream.readLong();
+        String startContinuationToken = Utils.readIntString(inputStream);
+        String endContinuationToken = Utils.readIntString(inputStream);
+        int index = inputStream.readInt();
+        int totalItems = inputStream.readInt();
+        String cacheSessionId = Utils.readIntString(inputStream);
+        cosmosChangeFeedFindToken =
+            new CosmosChangeFeedFindToken(bytesRead, startContinuationToken, endContinuationToken, index, totalItems,
+                cacheSessionId, version);
+        break;
+      default:
+        throw new IllegalStateException("Unknown version of cloud token: " + version);
+    }
+    return cosmosChangeFeedFindToken;
+  }
+
+  /**
+   * Serialize {@link CosmosChangeFeedFindToken} to byte array.
+   * @return serialized byte array.
+   */
+  public byte[] toBytes() {
+    byte[] buf = new byte[size()];
+    ByteBuffer bufWrap = ByteBuffer.wrap(buf);
+    bufWrap.putShort(getVersion());
```

Comment:
Please be consistent on using getters (getVersion()) vs straight reference (index, totalItems).  I prefer the latter.